pkgs/os-specific/linux/musl/default.nix

```diff
@@ -59,6 +59,10 @@ stdenv.mkDerivation rec {
   ];
   preConfigure = ''
     configureFlagsArray+=("--syslibdir=$out/lib")
+
```

Comment:
Please make this conditional on the nix level to avoid rebuilds and be easier to read. Also use `+=` in bash.